.circleci/config.yml

```diff
@@ -200,30 +206,41 @@ commands:
             echo "RELEASE_VERSION is: ${RELEASE_VERSION}"
 
   restore_dependency_cache:
-    # This logic cannot be done within the node orb currently. This is due to the fact that we use
-    # the package.json instead of the package-lock.json for our checksum.
     parameters:
       base-cache-path:
         type: string
         default: << pipeline.parameters.cache-version >>
     steps:
       - restore_cache:
           keys:
-            - << parameters.base-cache-path >>{{ checksum "package.json" }}
+            - << parameters.base-cache-path >>{{ checksum "package-lock.json" }}
 
   save_dependency_cache:
     parameters:
       base-cache-path:
         type: string
         default: << pipeline.parameters.cache-version >>
-      path-to-save:
-        type: string
-        default: 'node_modules'
     steps:
       - save_cache:
-          key: << parameters.base-cache-path >>{{ checksum "package.json" }}
+          key: << parameters.base-cache-path >>{{ checksum "package-lock.json" }}
           paths:
-            - << parameters.path-to-save >>
+            - node_modules
+            - packages/salesforcedx-apex-debugger/node_modules
+            - packages/salesforcedx-visualforce-language-server/node_modules
+            - packages/salesforcedx-vscode-apex-replay-debugger/node_modules
+            - packages/salesforcedx-vscode-soql/node_modules
+            - packages/salesforcedx-apex-replay-debugger/node_modules
+            - packages/salesforcedx-visualforce-markup-language-server/node_modules
+            - packages/salesforcedx-vscode-core/node_modules
+            - packages/salesforcedx-vscode-visualforce/node_modules
+            - packages/salesforcedx-sobjects-faux-generator/node_modules
+            - packages/salesforcedx-test-utils-vscode/node_modules
+            - packages/salesforcedx-vscode-apex/node_modules
+            - packages/salesforcedx-vscode-lightning/node_modules
+            - packages/system-tests/node_modules
+            - packages/salesforcedx-utils-vscode/node_modules
+            - packages/salesforcedx-vscode-apex-debugger/node_modules
+            - packages/salesforcedx-vscode-lwc/node_modules
```

Comment:
cache all the node_modules directories.  I saw this result in a bump from 1 m to 5 min to restore the cache, but it reduced the npm install time from 12+ m to 4